protocol/synthetix/contracts/modules/core/CollateralModule.sol

```diff
@@ -68,20 +68,23 @@ contract CollateralModule is ICollateralModule {
 
         Account.Data storage account = Account.load(accountId);
 
-        uint256 systemAmount = CollateralConfiguration
+        uint256 tokenAmountD18 = CollateralConfiguration
             .load(collateralType)
             .convertTokenToSystemAmount(tokenAmount);
 
         (uint256 totalDeposited, uint256 totalAssigned, uint256 totalLocked) = account
             .getCollateralTotals(collateralType);
 
-        // neither totalAssigned collateral nor totalLocked is available to be withdrawn
-        uint256 unavailableCollateral = totalAssigned > totalLocked ? totalAssigned : totalLocked;
-        if (totalDeposited - unavailableCollateral < systemAmount) {
-            revert InsufficientAccountCollateral(systemAmount);
+        // The amount that cannot be withdrawn from the protocol is the max of either
+        // locked collateral or delegated collateral.
+        uint256 unavailableCollateral = totalLocked > totalAssigned ? totalLocked : totalAssigned;
```

Comment:
inconsistant usage of D18 for local variables

Reply:
Yeah, it was previously called `systemAmount` which was a bit confusing because it evoked some kind of system quantity as opposed to the token amount that the user wants to withdraw.

And `tokenAmountSystemDenominated` sounds horrible. That's why I went with `tokenAmountD18`.